qiita_db/processing_job.py

```diff
@@ -186,13 +253,13 @@ def status(self):
             qdb.sql_connection.TRN.add(sql, [self.id])
             return qdb.sql_connection.TRN.execute_fetchlast()
 
-    @status.setter
-    def status(self, value):
+    def _set_status(self, value):
         """Sets the status of the job
 
         Parameters
         ----------
-        value : str, {'queued', 'running', 'success', 'error'}
+        value : str, {'queued', 'running', 'success', 'error',
```

Comment:
The call `new_status = qdb.util.convert_to_id(value, "processing_job_status")` at line 281 should already raise a meaningful error in case that it doesn't exist.